lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -3688,6 +3690,65 @@ void TFFunctionPartition::balanceRetainReleaseCount(SILValue oldResult,
   }
 }
 
+void TFFunctionPartition::insertReplacementHostOp(
+    ArrayRef<SILValue> resultValues) {
+  // Sanity check that all result values are tensor handles. Note SIL
+  // accelerator functions under the "tensorflow" convention can also return
+  // variant and resource tensors (in addition to tensor handles), but such
+  // functions will not generate all host-side code, and thus this method will
+  // not be called.
+  for (auto resultValue : resultValues) {
+    assert(
```

Comment:
is there already such a unit test that will reject these kinds of programs earlier on (like in type checker), so that we won't crash here?

Reply:
I copied this check from the other one. I think it is true.